RoaringBitmap/src/main/java/org/roaringbitmap/RoaringBitSet.java

```diff
@@ -0,0 +1,235 @@
+  }
+
+  @Override
+  public BitSet get(int fromIndex, int toIndex) {
+    BitSet bitSet = BitSetUtil.bitsetOf(roaringBitmap);
+    bitSet = bitSet.get(fromIndex, toIndex);
+    return new RoaringBitSet(fromBitSet(bitSet));
+  }
+
+  @Override
+  public int nextSetBit(int fromIndex) {
+    return (int) roaringBitmap.nextValue(fromIndex);
+  }
+
+  @Override
+  public int nextClearBit(int fromIndex) {
+    return (int) roaringBitmap.nextAbsentValue(fromIndex);
+  }
+
+  @Override
+  public int previousSetBit(int fromIndex) {
+    return (int) roaringBitmap.previousValue(fromIndex);
+  }
+
+  @Override
+  public int previousClearBit(int fromIndex) {
+    return (int) roaringBitmap.previousAbsentValue(fromIndex);
+  }
+
+  @Override
+  public int length() {
+    if (roaringBitmap.isEmpty()) {
+      return 0;
+    }
+    return roaringBitmap.last() + 1;
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return roaringBitmap.isEmpty();
+  }
+
+  @Override
+  public boolean intersects(BitSet set) {
+    if (set instanceof RoaringBitSet) {
+      return RoaringBitmap.intersects(roaringBitmap, ((RoaringBitSet) set).roaringBitmap);
+    }
+    return RoaringBitmap.intersects(roaringBitmap, fromBitSet(set));
+  }
+
+  @Override
+  public int cardinality() {
+    return roaringBitmap.getCardinality();
+  }
+
+  @Override
+  public void and(BitSet set) {
+    if (set instanceof RoaringBitSet) {
+      roaringBitmap.and(((RoaringBitSet) set).roaringBitmap);
+    } else {
+      roaringBitmap.and(fromBitSet(set));
+    }
+  }
+
+  @Override
+  public void or(BitSet set) {
+    if (set instanceof RoaringBitSet) {
+      roaringBitmap.or(((RoaringBitSet) set).roaringBitmap);
+    } else {
+      roaringBitmap.or(fromBitSet(set));
+    }
+  }
+
+  @Override
+  public void xor(BitSet set) {
+    if (set instanceof RoaringBitSet) {
+      roaringBitmap.xor(((RoaringBitSet) set).roaringBitmap);
+    } else {
+      roaringBitmap.xor(fromBitSet(set));
+    }
+  }
+
+  @Override
+  public void andNot(BitSet set) {
+    if (set instanceof RoaringBitSet) {
+      roaringBitmap.andNot(((RoaringBitSet) set).roaringBitmap);
+    } else {
+      roaringBitmap.andNot(fromBitSet(set));
+    }
+  }
+
+  @Override
+  public int hashCode() {
+    return roaringBitmap.hashCode();
+  }
+
+  @Override
+  public int size() {
+    if (roaringBitmap.isEmpty()) {
+      return 0;
+    }
+    int lastBit = Math.max(length(), 64);
+    int remainder = lastBit % 64;
+    return remainder == 0 ? lastBit : lastBit + 64 - remainder;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == this) {
+      return true;
+    }
+    if (obj instanceof RoaringBitSet) {
+      return roaringBitmap.equals(((RoaringBitSet) obj).roaringBitmap);
+    }
+    if (obj instanceof BitSet) {
+      return roaringBitmap.equals(fromBitSet((BitSet) obj));
+    }
+    return false;
+  }
+
+  @Override
+  public Object clone() {
+    super.clone();
```

Comment:
Probably not what you want to do here.

Reply:
Not that I'm not a committer and the fact that I read through the PR should not indicate that the feature is or is not considered to be a valid request on this project.

Reply:
fixed

Reply:
> Not that I'm not a committer and the fact that I read through the PR should not indicate that the feature is or is not considered to be a valid request on this project.

Thanks for you explain, it doesn't matter that whether this project accept or reject this PR.
 I did the feature in another project, but it's great that if the upstream project could accept it.